pallets/route-executor/src/lib.rs

```diff
@@ -575,6 +612,45 @@ impl<T: Config> Pallet<T> {
 		Ok(route)
 	}
 
+	pub fn disable_ed_handling_for_insufficient_assets(
+		skip_ed_disabling: &mut bool,
+		route_length: usize,
+		trade_index: usize,
+		trade: Trade<T::AssetId>,
+	) {
+		if route_length > 1
+			&& (!T::InspectRegistry::is_sufficient(trade.asset_in)
+				|| !T::InspectRegistry::is_sufficient(trade.asset_out))
+		{
+			*skip_ed_disabling = true;
+			//We optimize to set the state for middle trades only once at the first middle trade, then we change no state till the last trade
```

Comment:
does this really do anything? i do not see any performance gain, just complexity increase

Reply:
but there really shoudln't be any additional op as this item is never stored on chain, it is just used in memory during route processing

Reply:
i would rather annotate it somehow so the bechmark knows instead of trying to confuse it with more complex code

Reply:
ok, we can change it.
Yes, I know the real write won't happen but you have to do everything else storage vise and I read it's worth to not to do.

Reply:
all right, we will only add this overhead weight if insufficient asset is involved, already fixed.

Reply:
given the fact that the storage value is the cheapest to modify and here we talk about just one write, I don't see much value in complicating the already complex benchmark of route executor.

Reply:
I think only change in benchmarks is that we need trigger this branch of code by using insufficient asset. But this needs to be done anyway

I would say this will be always counted as 1 read and 1 write in benchmarks as:
```
* Multiple reads of the same value count as one read.
* Multiple writes of the same value count as one write.
* Multiple reads of the same value, followed by a write to that value, count as one read and one write.
* A write followed by a read only counts as one write.
```

My concern with performance was/is with storage overlay access as described here: 
https://substrate.stackexchange.com/a/2625

If we don't think complexity is worth it, we can do storage write on every iteration

Reply:
Yes, what I am doing is that I add the overhead in case of insufficient asset: 

https://github.com/galacticcouncil/hydration-node/blob/2396e44f731d42fcf9af2a4204ecd9d2a5001115/runtime/hydradx/src/assets.rs#L973-L978

As for killing the storage, it always incorporated. As the reading/writing to a storage value is the cheapest (comparing to other storage types), I think we are fine with the current solution. Let me know otherwise.

Reply:
we really have to use storage to have some sort of context stored during the execution, mby we could use some other way to share context without touching storage layer, but i am not aware of any currently. that would be good way to optimize this.